inc/dbmysql.class.php

```diff
@@ -970,8 +991,13 @@ public function tableExists($tablename, $usecache = true) {
       // with all known tables
       $retrieve_all = !$this->cache_disabled && empty($this->table_cache);
 
-      $result = $this->listTables($retrieve_all ? 'glpi_%' : $tablename);
       $found_tables = [];
+      $result = $this->listTables($retrieve_all ? 'glpi_%' : $tablename);
+      while ($data = $result->next()) {
+         $found_tables[] = $data['TABLE_NAME'];
+      }
+
+      $result = $this->listViews($retrieve_all ? 'glpi_%' : $tablename);
```

Comment:
As views will be used only on specific case, I think this second request should only be done if necessary, i.e.
```php
// Search in view only if not found in base tables or if $retrieve_all flag is active
if ($retrieve_all || !in_array($tablename, $found_tables)) {
   ...
}
```